samples/Azure.Management.Storage/Generated/Extensions/StorageAccountResourceExtensionClient.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.Threading;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Management.Storage;
+using Azure.ResourceManager;
+
+namespace Azure.Management.Storage.Mock
+{
+    /// <summary> A class to add extension methods to SubscriptionResource. </summary>
+    public partial class StorageAccountResourceExtensionClient : ArmResource
+    {
+        private ClientDiagnostics _storageAccountClientDiagnostics;
+        private StorageAccountsRestOperations _storageAccountRestClient;
+
+        /// <summary> Initializes a new instance of the <see cref="StorageAccountResourceExtensionClient"/> class for mocking. </summary>
+        protected StorageAccountResourceExtensionClient()
+        {
+        }
+
+        /// <summary> Initializes a new instance of the <see cref="StorageAccountResourceExtensionClient"/> class. </summary>
+        /// <param name="client"> The client parameters to use in these operations. </param>
+        /// <param name="id"> The identifier of the resource that is the target of operations. </param>
+        internal StorageAccountResourceExtensionClient(ArmClient client, ResourceIdentifier id) : base(client, id)
+        {
+        }
+
+        private ClientDiagnostics StorageAccountClientDiagnostics => _storageAccountClientDiagnostics ??= new ClientDiagnostics("Azure.Management.Storage.Mock", StorageAccountResource.ResourceType.Namespace, Diagnostics);
```

Comment:
Should we use `Azure.Management.Storage` here instead of `Azure.Management.Storage.Mock`?
```suggestion
        private ClientDiagnostics StorageAccountClientDiagnostics => _storageAccountClientDiagnostics ??= new ClientDiagnostics("Azure.Management.Storage", StorageAccountResource.ResourceType.Namespace, Diagnostics);
```

Reply:
This is fixed